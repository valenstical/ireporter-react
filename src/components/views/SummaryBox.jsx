import React from 'react';
import PropTypes from 'prop-types';

/**
 * The summary box component
 * @returns {object} The generated JSX object
 */
export default function SummaryBox({ summary }) {
  return (
    <header className="page-title">
      <div className="row" data-pg-name="Row">
        <div className="column column-xs-3" data-pg-name="Column" data-pg-collapsed>
          <div className="card card-sm summary">
            <i className="fa fa-file-text-o bg-status-draft" />
            <p>
              <small id="pending">{summary.pending}</small>
              <span>Pending</span>
            </p>
          </div>
        </div>
        <div className="column column-xs-3" data-pg-name="Column" data-pg-collapsed>
          <div className="card card-sm summary">
            <i className="fa bg-status-resolved fa-check" />
            <p>
              <small id="resolved">{summary.resolved}</small>
              <span>Resolved</span>
            </p>
          </div>
        </div>
        <div className="column column-xs-3" data-pg-name="Column" data-pg-collapsed>
          <div className="card card-sm summary">
            <i className="fa bg-status-rejected fa-remove" />
            <p>
              <small id="rejected">{summary.rejected}</small>
              <span>Rejected</span>
            </p>
          </div>
        </div>
        <div className="column column-xs-3" data-pg-name="Column" data-pg-collapsed>
          <div className="card card-sm summary">
            <i className="fa bg-status-pending fa-info" />
            <p>
              <small id="investigating">{summary.investigating}</small>
              <span title="Investigating">Investigating</span>
            </p>
          </div>
        </div>
      </div>
    </header>
  );
}

SummaryBox.propTypes = {
  summary: PropTypes.object.isRequired
};
