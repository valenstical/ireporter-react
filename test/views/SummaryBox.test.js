import React from 'react';
import { shallow } from 'enzyme';
import SummaryBox from '../../src/components/views/SummaryBox';

describe('<SummaryBox>', () => {
  it('should render without crashing when not active', () => {
    const wrapper = shallow(<SummaryBox summary={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
