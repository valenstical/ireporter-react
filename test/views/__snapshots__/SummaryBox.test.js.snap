// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryBox> should render without crashing when not active 1`] = `
<header
  className="page-title"
>
  <div
    className="row"
    data-pg-name="Row"
  >
    <div
      className="column column-xs-3"
      data-pg-collapsed={true}
      data-pg-name="Column"
    >
      <div
        className="card card-sm summary"
      >
        <i
          className="fa fa-file-text-o bg-status-draft"
        />
        <p>
          <small
            id="pending"
          />
          <span>
            Pending
          </span>
        </p>
      </div>
    </div>
    <div
      className="column column-xs-3"
      data-pg-collapsed={true}
      data-pg-name="Column"
    >
      <div
        className="card card-sm summary"
      >
        <i
          className="fa bg-status-resolved fa-check"
        />
        <p>
          <small
            id="resolved"
          />
          <span>
            Resolved
          </span>
        </p>
      </div>
    </div>
    <div
      className="column column-xs-3"
      data-pg-collapsed={true}
      data-pg-name="Column"
    >
      <div
        className="card card-sm summary"
      >
        <i
          className="fa bg-status-rejected fa-remove"
        />
        <p>
          <small
            id="rejected"
          />
          <span>
            Rejected
          </span>
        </p>
      </div>
    </div>
    <div
      className="column column-xs-3"
      data-pg-collapsed={true}
      data-pg-name="Column"
    >
      <div
        className="card card-sm summary"
      >
        <i
          className="fa bg-status-pending fa-info"
        />
        <p>
          <small
            id="investigating"
          />
          <span
            title="Investigating"
          >
            Investigating
          </span>
        </p>
      </div>
    </div>
  </div>
</header>
`;
